hal/src/stm32f2xx/interrupts_hal.c

```diff
@@ -80,8 +81,13 @@ static exti_state exti_saved_state = {0};
 
 /* Private function prototypes -----------------------------------------------*/
 
-void HAL_Interrupts_Attach(uint16_t pin, HAL_InterruptHandler handler, void* data, InterruptMode mode, HAL_InterruptExtraConfiguration* config)
+int HAL_Interrupts_Attach(uint16_t pin, HAL_InterruptHandler handler, void* data, InterruptMode mode, HAL_InterruptExtraConfiguration* config)
 {
+#if PLATFORM_ID == PLATFORM_ELECTRON_PRODUCTION
+  /* safety check that prevents users from attaching an interrupt to D7
+   * which is shared with BATT_INT_PC13 for power management */
+  if (pin == D7) return SYSTEM_ERROR_NOT_ALLOWED;
```

Comment:
Nitpicking: formatting.
```suggestion
  if (pin == D7) {
      return SYSTEM_ERROR_NOT_ALLOWED;
  }
```

Reply:
Fixed this formatting.